new generic function agglomerateByVariable

theotimeee opened the pull request:
Create a new generic function `agglomerateByVariable` to replace the methods `mergeCols` and `mergeRows`.
`agglomerateByVariable` takes parameter `MARGIN` to specify wether to run `mergeCols` or `mergeRows`

Commit:
deprecate mergeCols and mergeRows
add new function agglomerateByVariable

Commit:
up

Commit:
correct agglomerateByVariable

Commit:
correct indentation and lines lenghts in agglomerate.R, merge.R and deprecate.R

Commit:
up

Commit:
up

Commit:
change documentation

Commit:
up

Commit:
up

Commit:
up

Commit:
up

Commit:
Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByVariable

Commit:
correct agglomerate-methods man file

Commit:
up

Commit:
up

Commit:
correct use of mergeRefSeq parameter

Commit:
correct typo

TuomasBorman commented on R/agglomerate.R:
```diff
@@ -214,28 +287,63 @@ setMethod("agglomerateByRank", signature = c(x = "SummarizedExperiment"),
+#' @export
+setMethod("agglomerateByVariable", signature = c(x = "SummarizedExperiment"),
+            function(x, MARGIN, f, archetype = 1L, ...){
+                FUN <- switch(MARGIN, 
```
Use .check_MARGIN function to convert MARGIN to 1 or 2

TuomasBorman replied:
1 and 2 is general way in R to denote rows and cols.

TuomasBorman commented on R/deprecate.R:
```diff
@@ -38,3 +38,128 @@ setMethod("taxonomyTree", signature = c(x = "SummarizedExperiment"),
+                                        "Use agglomerateByVariable with 
+                                        parameter MARGIN = 'cols' instead"))
+                agglomerateByVariable(x, MARGIN = "cols", ...)
+            }
```
The indentation of this message is off. It shows in the console like this --> test all this deprecated functions
```

Warning message:
'mergeCols' is deprecated.Use agglomerateByVariable with 
                                        parameter MARGIN = 'cols' instead 
```

TuomasBorman commented on R/agglomerate.R:
```diff
@@ -263,11 +371,14 @@ setMethod("agglomerateByRank", signature = c(x = "SingleCellExperiment"),
-              .Deprecated(old="agglomerateByRank", new="mergeFeaturesByRank", "Now agglomerateByRank is deprecated. Use mergeFeaturesByRank instead.")
-              x <- agglomerateByRank(x, ..., altexp = altexp, strip_altexp = strip_altexp)
-              x
-          }
+            function(x, ..., altexp = NULL, strip_altexp = TRUE){
+                .Deprecated(old="agglomerateByRank", new="mergeFeaturesByRank",
+                            "Now agglomerateByRank is deprecated. 
+                            Use mergeFeaturesByRank instead.")
+                x <- agglomerateByRank(x, ..., altexp = altexp, 
+                                        strip_altexp = strip_altexp)
+                x
```
Move these to deprecate.R

These functions will be removed. only agglomerateByRank, agglomerateByPrevalence and agglomerateByVariable will stay

TuomasBorman commented:
Check those comments. I think this looks good

TuomasBorman committed:
Merge branch 'devel' into agglomerateByVariable

Commit:
changes from review

Commit:
merge devel and small changes

Merge branch 'agglomerateByVariable' of https://github.com/microbiome/mia into agglomerateByVariable

# Conflicts:
#	man/deprecate.Rd

Commit:
correct arguments deprecate file

TuomasBorman commented on R/agglomerate.R:
```diff
@@ -1,12 +1,19 @@
+#' Agglomerate or merge data using taxonomic information
 #'
 #' \code{agglomerateByRank} can be used to sum up data based on associations
 #' with certain taxonomic ranks, as defined in \code{rowData}. Only available
-#' \code{\link{taxonomyRanks}} can be used.
+#' \code{\link{taxonomyRanks}} can be used. 
+#' 
+#' \code{agglomerateByVariable} merges data on rows or columns of a 
+#' \code{SummarizedExperiment} as defined by a \code{factor} alongside the 
+#' chosen dimension. Metadata from the \code{rowData} or \code{colData} are 
+#' retained as defined by \code{archetype}.
+#' \code{\link[SummarizedExperiment:SummarizedExperiment-class]{assay}} are 
+#' agglomerated, i.e. summed up. If the assay contains values other than counts 
+#' or absolute values, this can lead to meaningless values being produced. 
```
User wonders what is the difference between *Rank and *Variable.

*Rank: the data is agglomerated by taking into account the whole taxonomy table. --> use if you want to agglomerate based on taxonomy rank.
*Variable: the data is agglomerated based on certain variable. --> use if you want to agglomerate based on some other variable

Try to rephrase that

Commit:
add explanation difference between *rank and *variable

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByVariable

# Conflicts:
#	NEWS
#	R/deprecate.R
#	man/deprecate.Rd

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByVariable

# Conflicts:
#	NEWS

Commit:
merge devel

Merge branch 'devel' of https://github.com/microbiome/mia into agglomerateByVariable

# Conflicts:
#	DESCRIPTION
#	NEWS
#	R/agglomerate.R
#	R/deprecate.R
#	man/agglomerate-methods.Rd
#	man/deprecate.Rd

Commit:
fix documentation deprecate file

antagomir commented:
Seems good! Conflicts to be resolved

TuomasBorman committed:
Sync

Sync

# Conflicts:
#	NEWS
#	R/deprecate.R
#	man/deprecate.Rd

TuomasBorman committed:
up
